fvm/meter/weighted_meter.go

```diff
@@ -265,38 +312,19 @@ type WeightedMeter struct {
 type WeightedMeterOptions func(*WeightedMeter)
 
 // NewMeter constructs a new Meter
-func NewMeter(computationLimit, memoryLimit uint, options ...WeightedMeterOptions) Meter {
-
+func NewMeter(params MeterParameters) Meter {
 	m := &WeightedMeter{
-		computationLimit:       uint64(computationLimit) << MeterExecutionInternalPrecisionBytes,
-		memoryLimit:            uint64(memoryLimit),
-		computationWeights:     DefaultComputationWeights,
-		memoryWeights:          DefaultMemoryWeights,
+		computationLimit:       uint64(params.ComputationLimit) << MeterExecutionInternalPrecisionBytes,
+		memoryLimit:            params.MemoryLimit,
+		computationWeights:     params.ComputationWeights,
+		memoryWeights:          params.MemoryWeights,
```

Comment:
maybe we can make params private and embed it instead? `MeterExecutionInternalPrecisionBytes` shift can be in `Default` and `WithComputationLimit` functions.

Reply:
i'll the fields private / embed the param, but will probably leave precision as a const for now